src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerLinkCase.java

```diff
@@ -1,135 +1,112 @@
     private boolean linkCaseEnabled;
 
-    private static final CcdPageConfiguration linkCaseSelectCase = new LinkCaseSelectCase();
+    @Autowired
+    CaseLinkedNotification caseLinkedNotification;
 
     @Override
     public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
         if (linkCaseEnabled) {
-            doConfigure(configBuilder);
-        }
-    }
+            new PageBuilder(configBuilder
+                .event(CASEWORKER_LINK_CASE)
+                .forStates(Submitted, CaseManagement, AwaitingHearing, AwaitingOutcome)
+                .name("Link cases")
+                .description("To link related cases")
+                .submittedCallback(this::submitted)
+                .grant(CREATE_READ_UPDATE, SUPER_USER,
+                    ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_HEARING_CENTRE_ADMIN, CASEWORKER_ADMIN_PROFILE,
+                    ST_CIC_HEARING_CENTRE_TEAM_LEADER)
+                .grantHistoryOnly(
+                    ST_CIC_CASEWORKER,
+                    ST_CIC_SENIOR_CASEWORKER,
+                    ST_CIC_HEARING_CENTRE_ADMIN,
+                    ST_CIC_HEARING_CENTRE_TEAM_LEADER,
+                    ST_CIC_SENIOR_JUDGE,
+                    SUPER_USER,
+                    ST_CIC_JUDGE))
+                .page("createCaseLink")
+                .pageLabel("Case Link")
+                .optional(CaseData::getCaseLinks, ALWAYS_HIDE, null, true)
+                .optional(CaseData::getLinkedCasesComponentLauncher,
+                    null, null, null, null, "#ARGUMENT(CREATE,LinkedCases)");
 
-    private void doConfigure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
-        PageBuilder pageBuilder = new PageBuilder(configBuilder
-            .event(CASEWORKER_LINK_CASE)
-            .forStates(Submitted, CaseManagement, AwaitingHearing, AwaitingOutcome)
-            .name("Links: Link case")
-            .showSummary()
-            .description("Links: Link case")
-            .aboutToSubmitCallback(this::aboutToSubmit)
-            .submittedCallback(this::submitted)
-            .grant(CREATE_READ_UPDATE,
-                ST_CIC_CASEWORKER, ST_CIC_SENIOR_CASEWORKER, ST_CIC_HEARING_CENTRE_ADMIN,
-                ST_CIC_HEARING_CENTRE_TEAM_LEADER));
-
-        addWarning(pageBuilder);
-        linkCaseSelectCase.addTo(pageBuilder);
+        }
     }
 
-    @SneakyThrows
-    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
-                                                                       CaseDetails<CaseData, State> beforeDetails) {
-        log.info("Caseworker link the case callback invoked for Case Id: {}", details.getId());
+    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
+                                               CaseDetails<CaseData, State> beforeDetails) {
         var data = details.getData();
```

Comment:
Let's not use var. Needs to be replaced with CaseData.

Reply:
Changed the type to CaseData